packages/datadog-instrumentations/src/kafkajs.js

```diff
@@ -81,6 +80,10 @@ addHook({ name: 'kafkajs', file: 'src/index.js', versions: ['>=1.4'] }, (BaseKaf
               }),
               innerAsyncResource.bind(err => {
                 if (err) {
+                  // disable header injection for this error (unfortunately the error name / type is not more specific)
```

Comment:
```suggestion
                  // Fixes bug where we would inject message headers for kafka brokers that don't support headers
                  // (version <0.11). On the error, we disable header injection.
                  // Tnfortunately the error name / type is not more specific.
                  // This approach is implemented by other tracers as well.


```

Reply:
Basically this needs the explanation that you have in the PR description

Reply:
done.